tunnel/intra/doh/doh.go

```diff
@@ -176,6 +178,7 @@ func NewTransport(rawurl string, addrs []string, dialer *net.Dialer, listener Li
 		listener: listener,
 		dialer:   dialer,
 		ips:      ipmap.NewIPMap(dialer.Resolver),
+		auth:     newClientAuthWrapper(loader),
```

Comment:
It looks like there's no need for this to be a field of `transport`.  Thanks to the magic of garbage-collected languages, it can just be a local variable.

Reply:
Good observation, I've added a bit of logic to make the TLS Config optional.